src/compiler/scala/tools/nsc/backend/jvm/PerRunInit.scala

```diff
@@ -9,10 +9,13 @@ import scala.collection.mutable.ListBuffer
  * The trait provides an `initialize` method that runs all initializers added through `perRunLazy`.
  */
 trait PerRunInit {
+  //We have to synchronize on inits, as many of the initialisers are themselves lazy,
+  // so the back end may initialise them in parallel, and ListBuffer is not threadsafe
+  // Not sure it is sensible to have a perRunInit that is itself a lazy val
```

Comment:
the idea is that only those `PerRunInit`s that are actually used are also created, and re-initialized on the next run.

Reply:
yes but we have lazy val which are per run init, so the registration of them is potentially in multiple threads, either at class load time or when the val is first touched. I am not sure why a per-run is lazy anyway, but I could not reason as to it being guaranteed to be threadsafe, and at best it would be fragile.

The problem is the actual lit f per-runs not in the actual target